pkg/orchestrator/housekeeping_test.go

```diff
@@ -0,0 +1,343 @@
+//go:build unit || !integration
+
+package orchestrator
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/bacalhau-project/bacalhau/pkg/jobstore"
+	"github.com/bacalhau-project/bacalhau/pkg/models"
+	"github.com/bacalhau-project/bacalhau/pkg/test/mock"
+	"github.com/stretchr/testify/suite"
+	"go.uber.org/mock/gomock"
+)
+
+const (
+	mockJobTimeout                 = 60              // 1 minute
+	timeoutBuffer                  = 1 * time.Minute // another minute. only executions older than 2 minutes are expired
+	notExpiredModifyTime           = -30 * time.Second
+	expiredWithinBufferModifyTime  = -90 * time.Second
+	expiredOutsideBufferModifyTime = -120 * time.Second
+)
+
+type HousekeepingTestSuite struct {
+	suite.Suite
+	ctrl                 *gomock.Controller
+	mockJobStore         *jobstore.MockStore
+	mockEvaluationBroker *MockEvaluationBroker
+	housekeeping         *Housekeeping
+}
+
+func (s *HousekeepingTestSuite) SetupTest() {
+	s.ctrl = gomock.NewController(s.T())
+	s.mockJobStore = jobstore.NewMockStore(s.ctrl)
+	s.mockEvaluationBroker = NewMockEvaluationBroker(s.ctrl)
+
+	h, _ := NewHousekeeping(HousekeepingParams{
+		EvaluationBroker: s.mockEvaluationBroker,
+		JobStore:         s.mockJobStore,
+		Interval:         200 * time.Millisecond,
+		Workers:          1,
+		TimeoutBuffer:    timeoutBuffer,
+	})
+
+	s.housekeeping = h
+}
+
+func (s *HousekeepingTestSuite) TearDownTest() {
+	s.housekeeping.Stop(context.Background())
+	s.ctrl.Finish()
+}
+
+func (s *HousekeepingTestSuite) TestHousekeepingTasks() {
+	var tests = []struct {
+		name                 string
+		ModifyTimes          []time.Time
+		expectedEnqueueCount int
+		jobCount             int
+		jobType              string
+		executionState       models.ExecutionStateType
+	}{
+		{
+			name: "TestExpiredExecutionOutsideBuffer",
```

Comment:
nit: including 'Test' in the name is redundant.

Reply:
renamed in https://github.com/bacalhau-project/bacalhau/pull/3974/commits/7a0b02e08cfa7910311c7db16b763fd962d1c68f